app/src/main/java/com/recipe/android/hereismycard/makeCard/MakeCardFragment.kt

```diff
@@ -0,0 +1,201 @@
+import com.recipe.android.hereismycard.R
+import com.recipe.android.hereismycard.databinding.FragmentMakeCardBinding
+import com.recipe.android.hereismycard.makeCard.adapter.SelectCardBgAdapter
+import euphony.lib.receiver.AcousticSensor
+import euphony.lib.receiver.EuRxManager
+import euphony.lib.transmitter.EuTxManager
+
+class MakeCardFragment : Fragment() {
+    private var _binding: FragmentMakeCardBinding? = null
+    private val binding get() = _binding!!
+    private var isSpeak = true
+    private var speakText = ""
+    private var saveBackgroundPosition = 0
+    lateinit var db: AppDatabase
+    lateinit var selectCardBgAdapter: SelectCardBgAdapter
+
+    private val mTxManager: EuTxManager by lazy {
+        EuTxManager()
+    }
+
+    private val mRxManager: EuRxManager by lazy {
+        EuRxManager()
+    }
+
+    private val requiredPermission = arrayOf(
+        android.Manifest.permission.RECORD_AUDIO
+    )
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        requestAudioPermission()
+        initDB()
+        _binding = FragmentMakeCardBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        selectCardBgAdapter = SelectCardBgAdapter()
+        selectCardBgAdapter.setItemClickListener(object : SelectCardBgAdapter.OnItemClickListener {
+            override fun onClick(v: View, position: Int) {
+                saveBackgroundPosition = position
+                binding.layoutCard.setBackgroundResource(bgList[position])
+            }
+        })
+        binding.rvSelectBg.apply {
+            adapter = selectCardBgAdapter
+            layoutManager = LinearLayoutManager(context, RecyclerView.HORIZONTAL, false)
+        }
+
+        binding.btnSend.setOnClickListener {
+            if (isSpeak) {
+                binding.btnSend.playAnimation()
+                stopListen()
+                playSpeak()
+                Toast.makeText(requireContext(), "SPEAK ON & LISTEN OFF", Toast.LENGTH_SHORT).show()
+            } else {
+                binding.btnSend.pauseAnimation()
+                playListen()
+                stopSpeak()
+                Toast.makeText(requireContext(), "SPEAK OFF & LISTEN ON", Toast.LENGTH_SHORT).show()
+            }
+        }
+    }
+
+    private fun playListen() {
+        mRxManager.listen()
+        makeCardData()
+    }
+
+    private fun makeCardData() {
+        mRxManager.acousticSensor = AcousticSensor { card ->
+            Log.i("RESULT_TEXT", card)
+            val split = card.split("/")
+            val name = split[0]
+            val tel = split[1]
+            val email = split[2]
+            val address = split[3]
+            val bgPosition = split[4].toInt()
```

Comment:
If you use this not-null string, 
the app crashes as shown in this picture.

![image](https://user-images.githubusercontent.com/66951780/129933208-5fd99033-4e77-42f7-a34e-4aedf68cc73e.png)

We also need to consider the case when the sender does not fill in the blanks, 
so why not fix it like the one below?

```suggestion
            var name = ""
            var tel = ""
            var email = ""
            var address = ""
            var bgPosition = 0
            card.split("/").forEachIndexed { idx, split ->
                when (idx) {
                    0 -> name = split
                    1 -> tel = split
                    2 -> email = split
                    3 -> address = split
                    4 -> bgPosition = split.toInt()
                }
            }
```

Reply:
It works well! 
Thanks!